src/Tasks/GenerateResource.cs

```diff
@@ -2432,6 +2432,12 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                     {
                         if (String.Compare(requestedAssemblyName.Name, candidateAssemblyName.Name, StringComparison.CurrentCultureIgnoreCase) == 0)
                         {
+                            string itemSpec = _assemblyFiles[i].ItemSpec;
+
+                            if (requestedAssemblyName.Name.Contains("System.Runtime"))
```

Comment:
There was a bug in the gist in my port s/peImageStart/resourceStart when reading resource length. @safern reported that it works for him after he applied that. I've updated the gist.